commons-numbers-core/src/main/java/org/apache/commons/numbers/core/DD.java

```diff
@@ -1924,6 +1924,18 @@ public DD pow(int n) {
         return computePow(x, xx, n);
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public boolean isZero() {
+        return equals(0.0, x) && equals(0.0, xx);
```

Comment:
The methods in this class assume a normalised representation such that `|x| >= |xx|`. Thus we only need to compare `x` to zero.

Due to the use of binary equality this method will not return true for `(-0.0, xx)`.

Reply:
Good hint. I found the -0.0 thing indeed confusing in the equals methods, but this is not the PR to change that. But surely isZero should return true for any kind of zero indeed :-). I'll fix it.

Reply:
fixed

Reply:
If you rebase on master then you will pick up changed I made to the DD and FP64 classes to use numerical equivalence for zero, i.e. `-0.0 == 0.0`. It fixes the usage of the classes as a field according to the field definition of zero.